search/tutorials/buildsearchbar.mdx

```diff
@@ -0,0 +1,207 @@
+---
+title: "Docs Search Quickstart"
+description: "A quick guideline to crawl docs and create a functional search component"
+---
+
+---
+
+### 1. Project Setup
+
+Let's create a new Next.js application and install the related packages:
+
+```shell
+npx create-next-app@latest search-docs-app
+cd search-docs-app
+npm install @upstash/search @upstash/search-ui lucide-react
+```
+
+---
+
+### 2. Add Environment Variables
+
+Find the environment variables from your database dashboard and add them to your `.env` file:
+
+```bash
+NEXT_PUBLIC_UPSTASH_SEARCH_URL=<YOUR_SEARCH_REST_URL>
+NEXT_PUBLIC_UPSTASH_SEARCH_READONLY_TOKEN=<YOUR_SEARCH_READONLY_TOKEN>
+```
+
+---
+
+### 3. Create the Component
+
+Create the search component in `app/components/search-bar.tsx`:
+
+```typescript title="app/components/search-bar.tsx"
+"use client"
+
+import { SearchBar } from "@upstash/search-ui"
+import "@upstash/search-ui/dist/index.css"
+
+import { Search } from "@upstash/search"
+import { FileText } from "lucide-react"
+
+const client = new Search({
+  url: process.env.NEXT_PUBLIC_UPSTASH_SEARCH_URL! ,
+  token: process.env.NEXT_PUBLIC_UPSTASH_SEARCH_READONLY_TOKEN!,
+})
+
+// 👇 your search index name
+const index = client.index<{ title: string }>("default")
+
+export default function SearchComponent() {
+  return (
+    <div className="max-w-sm mt-24 mx-auto">
+      <SearchBar.Dialog>
+        <SearchBar.DialogTrigger placeholder="Search..." />
+
+        <SearchBar.DialogContent>
+          <SearchBar.Input placeholder="Type to search..." />
+          <SearchBar.Results
+            searchFn={(query) => {
+              // 👇 100% type-safe: whatever you return here is
+              // automatically typed as `result` below
+              return index.search({ query, limit: 10, reranking: true })
+            }}
+          >
+            {(result) => (
+                <div key={result.id} onClick={() => {
+                   window.open(result.metadata?.url as string, "_blank")
+                 }}>
+               <SearchBar.Result 
+                 value={result.id} 
+                 key={result.id}
+                 className="cursor-pointer hover:bg-gray-50 transition-colors duration-200"
+               >
+                <SearchBar.ResultIcon>
+                    <FileText className="text-gray-600" />
+                </SearchBar.ResultIcon>
+
+                <SearchBar.ResultContent>
+                  <SearchBar.ResultTitle>
+                    {result.content.title}
+                  </SearchBar.ResultTitle>
+                  <p className="text-xs text-gray-500 mt-0.5">Docs</p>
+                </SearchBar.ResultContent>
+              </SearchBar.Result>
+              </div>
+            )}
+          </SearchBar.Results>
+        </SearchBar.DialogContent>
+      </SearchBar.Dialog>
+    </div>
+  )
+}
+```
+
+---
+
+### 4. Crawl Docs to Feed the Component
+
+Call `npx @upstash/search-crawler` in your command line and follow the CLI,
```

Comment:
we can add links to the docs https://upstash.com/docs/search/tools/databasemigrator or the repo of the crawler